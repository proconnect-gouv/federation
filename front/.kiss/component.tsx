import React from 'react';

interface DumbComponentProps {
  className: string;
}

export const DumbComponent: React.FC<DumbComponentProps> = React.memo(
  ({ className }: DumbComponentProps) => {
    return <div />;
  },
);

DumbComponent.displayName = 'DumbComponent';
