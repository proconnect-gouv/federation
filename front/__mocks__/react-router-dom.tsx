const actualModule = jest.requireActual('react-router-dom');

module.exports = {
  ...actualModule,
  Link: jest.fn(({ children }) => <span data-mockid={'Link'}>{children}</span>),
  NavLink: jest.fn(({ children }) => <div data-mockid={'NavLink'}>{children}</div>),
  Navigate: jest.fn(() => <div>ReactRouterDom Navigate Component</div>),
  Outlet: jest.fn(() => <div>ReactRouterDom Outlet Component</div>),
  Route: jest.fn(({ children, element }) => {
    if (element) {
      return element;
    }
    return <div data-mockid={'Route'}>{children}</div>;
  }),
  BrowserRouter: jest.fn(({ children, element }) => {
    if (element) {
      return element;
    }
    return <div data-mockid={'BrowserRouter'}>{children}</div>;
  }),
  Routes: jest.fn(({ children }) => children),
  matchPath: jest.fn(),
  useLocation: jest.fn(() => ({
    pathname: '',
  })),
  useNavigate: jest.fn(() => jest.fn()),
  useParams: jest.fn(),
  useSearchParams: jest.fn(() => [new URLSearchParams(), jest.fn()]),
};

export {};
