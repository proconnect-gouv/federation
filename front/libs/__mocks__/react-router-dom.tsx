const actualModule = jest.requireActual('react-router-dom');

module.exports = {
  ...actualModule,
  Link: jest.fn(({ children }) => <div>{children}</div>),
  NavLink: jest.fn(({ children }) => <div>{children}</div>),
  Redirect: jest.fn(() => <div>Redirect</div>),
  Route: jest.fn(({ children, render, ...rest }) => {
    if (render) {
      return render({ ...rest });
    }
    return <div>{children}</div>;
  }),
  Switch: jest.fn(({ children }) => children),
  matchPath: jest.fn(),
  useHistory: jest.fn(() => ({
    push: jest.fn(),
  })),
  useLocation: jest.fn(() => ({
    pathname: '',
  })),
};

export {};
