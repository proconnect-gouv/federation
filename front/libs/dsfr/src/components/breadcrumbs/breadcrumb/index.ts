/* istanbul ignore file */

// declarative file
export * from './breadcrumb.component';
