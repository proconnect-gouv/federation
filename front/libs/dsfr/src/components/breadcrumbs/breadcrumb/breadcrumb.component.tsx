import { Link } from 'react-router-dom';

import type { NavigationLinkInterface } from '@fc/common';

interface BreadCrumbComponentProps extends NavigationLinkInterface {
  isCurrent?: boolean;
}

export const BreadCrumbComponent = ({
  href,
  isCurrent = false,
  label,
  title,
}: BreadCrumbComponentProps) => (
  <li>
    <Link
      aria-current={isCurrent ? 'page' : undefined}
      className="fr-breadcrumb__link"
      title={title}
      to={href}>
      {label}
    </Link>
  </li>
);

BreadCrumbComponent.displayName = 'BreadCrumbComponent';
