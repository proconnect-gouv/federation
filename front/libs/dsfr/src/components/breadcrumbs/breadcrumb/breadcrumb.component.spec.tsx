import { render } from '@testing-library/react';
import { Link } from 'react-router-dom';

import { BreadCrumbComponent } from './breadcrumb.component';

describe('BreadCrumbComponent', () => {
  it('should match snapshot when is not current page', () => {
    // When
    const { container } = render(
      <BreadCrumbComponent href="any-href-mock" label="any-label-mock" />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(Link).toHaveBeenCalledOnce();
    expect(Link).toHaveBeenCalledWith(
      {
        children: 'any-label-mock',
        className: 'fr-breadcrumb__link',
        to: 'any-href-mock',
      },
      {},
    );
  });

  it('should match snapshot when is current page', () => {
    // When
    const { container } = render(
      <BreadCrumbComponent isCurrent href="any-href-mock" label="any-label-mock" />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(Link).toHaveBeenCalledOnce();
    expect(Link).toHaveBeenCalledWith(
      {
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'aria-current': 'page',
        children: 'any-label-mock',
        className: 'fr-breadcrumb__link',
        to: 'any-href-mock',
      },
      {},
    );
  });

  it('should match snapshot when title is defined', () => {
    // When
    const { container } = render(
      <BreadCrumbComponent href="any-href-mock" label="any-label-mock" title="any-title-mock" />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(Link).toHaveBeenCalledOnce();
    expect(Link).toHaveBeenCalledWith(
      {
        children: 'any-label-mock',
        className: 'fr-breadcrumb__link',
        title: 'any-title-mock',
        to: 'any-href-mock',
      },
      {},
    );
  });
});
