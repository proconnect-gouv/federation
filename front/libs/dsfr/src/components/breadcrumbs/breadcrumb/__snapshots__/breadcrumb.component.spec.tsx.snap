// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbComponent should match snapshot when is current page 1`] = `
<div>
  <li>
    <span
      data-mockid="Link"
    >
      any-label-mock
    </span>
  </li>
</div>
`;

exports[`BreadCrumbComponent should match snapshot when is not current page 1`] = `
<div>
  <li>
    <span
      data-mockid="Link"
    >
      any-label-mock
    </span>
  </li>
</div>
`;

exports[`BreadCrumbComponent should match snapshot when title is defined 1`] = `
<div>
  <li>
    <span
      data-mockid="Link"
    >
      any-label-mock
    </span>
  </li>
</div>
`;
